browser/ui/tabs/test/BUILD.gn

```diff
@@ -24,3 +24,40 @@ source_set("browser_tests") {
     "//content/test:test_support",
   ]
 }
+
+source_set("unit_tests") {
+  testonly = true
+
+  defines = [ "HAS_OUT_OF_PROC_TEST_RUNNER" ]
+
+  sources = [
+    "split_view_browser_data_unittest.cc",
+    "split_view_tab_strip_model_adapter_unittest.cc",
+  ]
+
+  deps = [
+    ":test_tab_strip_model_delegate",
+    "//brave/browser/ui/tabs",
+    "//chrome/browser",
+    "//chrome/browser/ui",
+    "//chrome/test:test_support",
+    "//chrome/test:test_support_ui",
+    "//components/tab_groups",
+  ]
+}
+
+# In order to resolve circular deps to the test_tab_strip_model_delegate,
+# declares source set here instead of adding these files to //brave/test/BUILD.gn
+source_set("test_tab_strip_model_delegate") {
+  sources = [
+    "//chrome/browser/ui/tabs/test_tab_strip_model_delegate.cc",
+    "//chrome/browser/ui/tabs/test_tab_strip_model_delegate.h",
+  ]
+
+  deps = [
+    "//base",
+    "//chrome/browser/extensions",
+    "//chrome/browser/ui",
+    "//components/tab_groups",
+  ]
+}
```

Comment:
@goodov Could you take a look at this part? When I added the `test_tab_strip_model_delegate` to the `test/build.gn` I needed to add `check_include = false` in the `tabs/test:unittest`. I thought it was due to circular deps like we have other places. Would this be proper way to solve that?

Reply:
https://github.com/brave/brave-core/pull/22772#discussion_r1542130372

Reply:
yeah, thanks! that's the right way of fixing this. Maybe I'd call this target `test_support` to make it more universal and allow other similar things to be included here in future.

we definitely don't want to introduce new `check_include = false`.

Reply:
Thanks a lot. `test_support` is so much better. Changed.